core/src/impl/Kokkos_FunctorAnalysis.hpp

```diff
@@ -90,6 +90,28 @@ struct DeduceFunctorPatternInterface<ParallelScanWithTotal<
   using type = FunctorPatternInterface::SCAN;
 };
 
+#if defined(__CUDA_ARCH__)
+template <typename T>
+KOKKOS_INLINE_FUNCTION void volatile_preload(const volatile T* src,
+                                             size_t count) {
+  // Ensure that subsequent code in this thread that may do
+  // non-`volatile` loads through `src` will not see stale
+  // non-coherent values in the cache by forcing `volatile` loads to
+  // precede it
+  auto ct         = reinterpret_cast<const volatile char*>(src);
+  unsigned char c = 0;
+  for (size_t i = 0; i < count * sizeof(T); ++i) {
+    c += ct[i];  // I'd rather avoid the addition, but this quiets the unused
+                 // variable warning about `c`
```

Comment:
What were you doing before?
```
c = ct[i]
```
and the usual `(void) c` wasn't enough?

Reply:
Yeah, nvcc is picky.

Though I assume this is ultimately going to be a 4.0 change, and we'll require C++17, and I can mark it with `[[maybe_unused]]`

Reply:
I found a seemingly better solution, of just `return c;`